src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net46.cs

```diff
@@ -14,14 +14,14 @@ public partial class Activity
         /// Gets or sets the current operation (Activity) for the current thread.  This flows
         /// across async calls.
         /// </summary>
-        public static Activity Current
+        public static Activity? Current
         {
             get { return s_current.Value; }
             set
             {
-                if (ValidateSetCurrent(value))
+                if (ValidateSetCurrent(value!))
```

Comment:
`ValidateSetCurrent` should actually accept nullable, if it is null, it would return true. Same goes for `SetCurrent`

Reply:
That is true but this API is confusing, it throws on null (not exact throwing, kind of notifying). There is many similar methods with such behavior. So as I was not sure if I should make it nullable or not I kept it nullable to get you guys opinions, will update accordingly all of them 
```csharp
private static bool ValidateSetCurrent(Activity activity)
{
    bool canSet = activity == null || (activity.Id != null && !activity.IsFinished);
    if (!canSet)
    {
        NotifyError(new InvalidOperationException("Trying to set an Activity that is not running"));
    }

    return canSet;
}
```

Reply:
actually it throws when `activity != null && activity.Id != null && !activity.IsFinished` so if `activity == null` `canSet` will be true hence it will not throw and return true. It is confusing because the if that throws is a negation of the actual condition which is an or, so you have to invert the condition.